### PR TITLE
Simplify CI by removing unused Docker images

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title.

These images are not unused by the project.
This PR proposes that CI is trimmed down with the intent of simplifying it.

cc @carmocca @borda

### CODE REVIEW

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -159,16 +159,9 @@ subprojects:
       - "!*.md"
       - "!**/*.md"
     checks:
-      - "build-cuda (3.9, 1.11, 11.3.1)"
-      - "build-cuda (3.9, 1.12, 11.6.1)"
-      - "build-cuda (3.9, 1.13, 11.7.1)"
       - "build-cuda (3.10, 2.0, 11.7.1)"
       - "build-ipu (3.8, 1.13)"
       - "build-NGC"
```
Comment (carmocca) [inline .github/checkgroup.yml:164]: @Borda Is the NGC image still used anywhere? Could we drop it too?

**Review by awaelchli — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
We shall keep the production PL images, fixed in #17353

**Comment by carmocca** [comment]
Closing since Jirka doesn't agree with this

**Comment by Borda** [comment]
> Closing since Jirka doesn't agree with this



> Closing since Jirka doesn't agree with this

If the issue is with the number of image, I am fine to prune them, but let's keep at least one per release

**Pull request closed without merging**